### DISCUSSION TITLE
Are there an option to not build a new apk for each test-file

### DISCUSSION THREAD

**Discussion by benjamin-inspectly** [leancodepl/patrol#1413] [Q&A] [answered]
Hi!
Maybe it's the structure of my tests, but I'm having only 2 "xyz_test.dart" files at the moment, and I can see that it builds my APK for each file separately and runs the tests afterwards..
I don't see the reason for this and was wondering if I was doing something wrong?

Did I miss something in the documentation about this? Would appreciate if you could point me in the right direction - thank you :)

**Comment by bartekpacia** [comment, marked as answer]
Hi Benjamin :)

The problem you're talking about is the shortcoming of the integration_test plugin. I [reported it](https://github.com/flutter/flutter/issues/115751) a few months ago.

---

Fortunately, today we released Patrol 2.0, which has the **test bundling** feature. It fixes exactly the problem you're having – you build the app once (with all tests _bundled_ together), and then run all the tests.

Learn more:
- [a Twitter thread about 2.0](https://twitter.com/LeanCodePl/status/1673641399990845442)
- [technical article explaining how the **test bundling** feature works](https://leancode.co/blog/patrol-2-0-improved-flutter-ui-testing)
- [migration guide to 2.0](https://patrol.leancode.co/v2)

**Reply by benjamin-inspectly** [reply]
Ahh brilliant! Thank you so much for the answer! :)
I'll give it a try :) 👍
